server/src/test/java/se/fortnox/reactivewizard/server/RwServerGzipTest.java

```diff
@@ -0,0 +1,126 @@
+package se.fortnox.reactivewizard.server;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import se.fortnox.reactivewizard.ExceptionHandler;
+import se.fortnox.reactivewizard.RequestHandler;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RwServerGzipTest {
+    private RwServer rwServer;
+
+    @Test
+    public void shouldCompressAllowedContentTypes() {
+        Arrays.asList("text/plain", "application/xml", "text/css", "application/x-javascript", "application/json")
```

Comment:
Extend with examples that include directives: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type